### PR TITLE
notify: add email integration tests

### PR DESCRIPTION
**Pull request by simonpasquier**
Given the last issue with the email receiver (#1721), I felt that we needed to improve our test coverage. I've been trying  different mail catchers, none are perfect but [MailDev](https://github.com/djfarrelly/MailDev) is pretty decent. It supports authentication (only LOGIN and not with STARTTLS), but it doesn't support SMTPS.

These tests aren't executed by default as you need to provide the addresses of the servers but I've documented the procedure in the code. Of course, I've updated Circle CI to run them automatically.

Related to #1517.

### CODE REVIEW

File: notify/email.go [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+	buffer := &bytes.Buffer{}
+	multipartWriter := multipart.NewWriter(buffer)
+
+	fmt.Fprintf(wc, "Date: %s\r\n", time.Now().Format(time.RFC1123Z))
+	fmt.Fprintf(wc, "Content-Type: multipart/alternative;  boundary=%s\r\n", multipartWriter.Boundary())
+	fmt.Fprintf(wc, "MIME-Version: 1.0\r\n")
+
+	// TODO: Add some useful headers here, such as URL of the alertmanager
+	// and active/resolved.
+	fmt.Fprintf(wc, "\r\n")
```
Comment (mxinden) [inline notify/email.go]: Does this ignore a possible write error? Not sure if this is just a buffer or a connection.

Comment (simonpasquier) [reply notify/email.go]: Yes and the same with the other `fmt.Fprintf(wc, ...)` statements above.

File: notify/email_test.go [hunk 2]
```diff
@@ -0,0 +1,450 @@
...
+// Some tests require a running mail catcher. We use MailDev for this purpose,
+// it can work without or with authentication (LOGIN only). It exposes a REST
+// API which we use to retrieve and check the sent emails.
+//
+// Those tests are only executed when specific environment variables are set,
+// otherwise they are skipped. The tests must be run by the CI.
+//
+// To run the tests locally, you should start 2 MailDev containers:
+//
+// $ docker run --rm -p 1080:1080 -p 1025:1025 --entrypoint bin/maildev djfarrelly/maildev -v
```
[the remark this answers is not in the document]
Comment (simonpasquier) [reply notify/email_test.go]: I've just checked and I now remember why I didn't pick an exact version: `latest` is one year old and the one before (`1.0.0-rc2`) is 2 years old.

https://hub.docker.com/r/djfarrelly/maildev/tags

[the remark this answers is not in the document]
Comment (mxinden) [reply notify/email_test.go]: How about pinning a particular docker image sha instead then?

**Pull request merged**